### PR TITLE
Buildbot fixes

### PR DESCRIPTION
**Pull request by dmaluenda**
Fixed test:
 - CTFconsensus
 - ConsenusPicking
 - MaxShift
 - TriggerData

### CODE REVIEW

File: xmipp3/protocols/protocol_particle_pick_consensus.py [hunk 1]
```diff
@@ -228,10 +228,12 @@ def _loadOutputSet(self, SetClass, baseName):
             outputSet.setStreamState(outputSet.STREAM_OPEN)
             outputSet.setBoxSize(self.getMainInput().getBoxSize())
 
-        #inMicsPointer = Pointer(self.getMapper().getParent(
-        #                                    self.getMainInput().getMicrographs()),
-        #                                    extended='outputMicrographs')
-        inMicsPointer = Pointer(self.getMainInput().getMicrographs())
+        # FIXME: The line below should work but it fails in streaming when extracting particles
+        # inMicsPointer = self.getMainInput().getMicrographs()
+        # FIXME: The line below fix the error in streaming due to the line above.
```
Comment (DStrelak) [inline xmipp3/protocols/protocol_particle_pick_consensus.py:233]: too many fixme :)

File: xmipp3/protocols/protocol_trigger_data.py [hunk 2]
```diff
@@ -147,12 +147,13 @@ def _checkNewOutput(self):
...
-        # If finished if:      - in non-streaming mode the output is released
-        self.finished = ( (not self.allImages and
-                           len(self.images) > self.outputSize)
-                          or # - in streaming if the input is closed and all done
-                          (self.allImages and self.streamClosed and
-                           len(self.images) == len(self.imsSet)) )
+        if self.streamClosed:
+            self.finished = True
+        elif not self.allImages.get():
+            self.finished = len(self.images) > self.outputSize
```
Comment (DStrelak) [inline xmipp3/protocols/protocol_trigger_data.py:153]: `>=` ?

File: xmipp3/protocols/protocol_trigger_data.py [hunk 3]
```diff
@@ -170,7 +171,7 @@ def _checkNewOutput(self):
     def _fillingOutput(self):
         imsSqliteFn = '%s.sqlite' % self.getImagesType('lower')
         outputName = 'output%s' % self.getImagesType()
-        if len(self.images) >= self.outputSize:
+        if len(self.images) >= self.outputSize or self.finished:
```
Comment (DStrelak) [inline xmipp3/protocols/protocol_trigger_data.py:174]: `self.finished` is set on line 153, i.e. left expression is equivalent to right one

File: xmipp3/protocols/protocol_trigger_data.py [hunk 4]
```diff
@@ -240,12 +241,41 @@ def _updateOutputSet(self, outputName, outputSet, state=Set.STREAM_OPEN):
 
     # --------------------------- INFO functions ------------------------------
     def _summary(self):
+
+        input = self.inputImages.get()
+        if input is None:
+            imagesType = "(or not ready)"
```
Comment (DStrelak) [inline xmipp3/protocols/protocol_trigger_data.py:247]: this seems strange

File: xmipp3/protocols/protocol_trigger_data.py [hunk 5]
```diff
@@ -240,12 +241,41 @@ def _updateOutputSet(self, outputName, outputSet, state=Set.STREAM_OPEN):
...
+        else:
+            imagesType = input.getClassName().split("SetOf")[1]
         summary = []
-        imagesType = self.inputImages.get().getClassName().split("SetOf")[1]
-        if not hasattr(self, 'output%s'%imagesType):
-            summary.append("Not enough images for output yet.")
+
+        if self.allImages.get() and not self.splitImages.get():
+            summary.append("MODE: *full streaming*.")
+            triggeredMsg = ("'output%s' released, it will be growing up "
```
Comment (DStrelak) [inline xmipp3/protocols/protocol_trigger_data.py:254]: `output %s` (multiple occurrences)?

File: xmipp3/protocols/protocol_trigger_data.py [hunk 6]
```diff
@@ -240,12 +241,41 @@ def _updateOutputSet(self, outputName, outputSet, state=Set.STREAM_OPEN):
...
+        if self.getOutputsSize():
+            summary.append(triggeredMsg)
+        else:
+            summary.append("Not enough input %s to release an output, yet."
+                           % imagesType)
+            summary.append("At least, %d items are needed to trigger an output."
+                           % self.outputSize.get())
+
+        if (self.isFinished() and
+                self.outputSize.get() > [o for o in
```
Comment (DStrelak) [inline xmipp3/protocols/protocol_trigger_data.py:275]: `self.finished`? doesn't this make the first expression same as the second one?

File: xmipp3/protocols/protocol_trigger_data.py (hunk 4 above)
Comment (dmaluenda) [reply xmipp3/protocols/protocol_trigger_data.py]: refactored

File: xmipp3/protocols/protocol_trigger_data.py (hunk 5 above)
Comment (dmaluenda) [reply xmipp3/protocols/protocol_trigger_data.py]: refactored

**Pull request merged**